### PR TITLE
Add 'include_path' property for include files

### PR DESCRIPTION
**Pull request by tudortimi**

### CODE REVIEW

**Comment by tudortimi** [comment]
This is currently a dummy PR to show my intention of working on #158

**Comment by tudortimi** [comment]
I'm thinking it would be better to throw the exception for illegal include path in the `Edatool` constructor.

**Comment by tudortimi** [comment]
Ideally, I think it would be better to use a `File` class instead of a dictionary, which can do the required validations in its constructor. This would be a major API change, though. We could introduce it internally in the `Edatool` constructor:

```
class Edatool(object):
    def __init__(self, edam=None, work_root=None, eda_api=None):
        # ...
        for file in files:
            self.files.append(File(edam.get('files', []))
```

**Comment by tudortimi** [comment]
If we merge #160, the validation of `include_path` can be moved there.

File: edalize/edatool.py [hunk 1]
```diff
@@ -268,6 +271,10 @@ def __init__(self, name, file_type, logical_name):
         for f in self.files:
             if 'is_include_file' in f and f['is_include_file']:
                 _incdir = os.path.dirname(f['name']) or '.'
+                if 'include_path' in f and f['include_path']:
+                    if os.path.isabs(f['include_path']):
+                        raise EdamException("'include_path' of file {} cannot be absolute".format(f))
```
Comment (olofk) [inline edalize/edatool.py:276]: I think we should allow absolute paths for now. There might be cases where users need to include some system library. It's not portable, but at least we wouldn't prevent the use from doing it. Maybe we could print a warning instead

**Comment by olofk** [comment]
Thanks. Looks great, but I would prefer to drop the check for absolute paths

**Comment by tudortimi** [comment]
The check exists mostly to show that I only thought about what to do with relative paths. What should be the behavior for absolute paths? The same `os.path.relpath` call?

**Comment by tudortimi** [comment]
I removed the check for absolute paths.

**Comment by olofk** [comment]
Thanks! I think this should do it

**Comment by olofk** [comment]
@tudortimi FYI, I realized now that there was some misunderstanding, so I need to change this a bit. We can't use relpath, because that doesn't work with paths that start with `..`, which is the case for all paths generated by FuseSoC (e.g. `../src/core_name/subdir/file.v`). The following spec will be implemented

In the EDAM file, `include_path` will be a path relative to the working directory (this is consistent with how file paths are currently specified)

In the FuseSoC CAPI2 core files, `include_path` will be a path relative to the core root.

**Pull request merged**